src/store/modules/cart.js

```diff
@@ -169,14 +172,29 @@ const actions = {
     commit(TYPES.SHOPPING_CART_REPLACE_ALL_BOOK_PRODUCT, {});
     dispatch('saveBookProductShoppingCart');
   },
-  saveBookProductShoppingCart({ state }) {
+  async saveBookProductShoppingCart({ state, getters }) {
     saveShoppingCartToStorage(state.shoppingCartBookProductByIdMap, 'book');
+    if (getters.loginAddress) {
+      const cart = Object.values(state.shoppingCartBookProductByIdMap);
+      try {
+        if (cart.length) {
+          await this.$api.$post(postShoppingCart(), { cart });
+        } else {
+          await this.$api.$post(postShoppingCart(), { cart });
```

Comment:
why they are executing the same API ?
```suggestion
          await this.$api.$delete(deleteShoppingCart());
```